tests/ui/type-alias-impl-trait/nested_in_closure.rs

```diff
@@ -5,6 +5,5 @@ fn main() {
         type Tait = impl Sized;
         let y: Tait = ();
         //~^ ERROR: item constrains opaque type that is not in its signature
```

Comment:
This error is for `main`, because `main` is not allowed to register hidden types for TAITs declared in its closures, but that isn't really avoidable in the current setup where the closure and its owner share an `InferCtxt` in typeck.

Reply:
Unfortunately the only thing we can do here is to allow functions to use the taits of their nested closures as if the taits were written within the function body directly. Scoping only works for naming the taits, not for obtaining them via inference.